app/src/main/java/com/iron/espresso/presentation/home/mystudy/studydetail/chat/ChattingViewModel.kt

```diff
@@ -0,0 +1,135 @@
+package com.iron.espresso.presentation.home.mystudy.studydetail.chat
+
+import androidx.hilt.Assisted
+import androidx.hilt.lifecycle.ViewModelInject
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.SavedStateHandle
+import com.iron.espresso.AuthHolder
+import com.iron.espresso.Logger
+import com.iron.espresso.UserHolder
+import com.iron.espresso.base.BaseViewModel
+import com.iron.espresso.domain.repo.ChatRepository
+import com.iron.espresso.ext.networkSchedulers
+import com.iron.espresso.ext.plusAssign
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+import java.net.SocketException
+import java.text.SimpleDateFormat
+import java.util.*
+
+class ChattingViewModel @ViewModelInject constructor(
+    @Assisted
+    private val savedState: SavedStateHandle,
+    private val chatRepository: ChatRepository
+) : BaseViewModel() {
+
+    private val studyId: Int by lazy {
+        savedState.get(KEY_STUDY_ID) ?: -1
+    }
+
+    private val _chatList = MutableLiveData<List<ChatItem>>()
+    val chatList: LiveData<List<ChatItem>> get() = _chatList
+
+    private var disposable: Disposable? = null
+
+    fun setup() {
+        compositeDisposable += chatRepository.setChat(studyId)
+            .networkSchedulers()
+            .subscribe({
+                getAllChats()
+            }, {
+                Logger.d("$it")
+            })
+    }
+
+    private fun getAllChats() {
+        disposable = chatRepository.getAll(studyId)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({
+                _chatList.value = groupByDate(ChatItem.of(it))
+            }, {
+                Logger.d("$it")
+            })
+    }
+
+    private fun groupByDate(chatList: List<ChatItem>): MutableList<ChatItem> {
```

Comment:
```suggestion
    private fun groupByDate(chatList: List<ChatItem>): List<ChatItem> {
```
별다른 이유가 없다면, List 타입으로 반환하는게 일반적으로 좋을 것 같아요.

Reply:
수정하였습니다